File: java_solutions_1/553.optimal-division.java
/*
 * @lc app=leetcode id=553 lang=java
 *
 * [553] Optimal Division
 */

class Solution {
    public String optimalDivision(int[] nums) {
        return nums.length == 1 ? String.valueOf(nums[0]) : nums.length == 2 ? nums[0] + "/" + nums[1] : nums[0] + "/(" + java.util.Arrays.stream(nums).skip(1).mapToObj(String::valueOf).collect(java.util.stream.Collectors.joining("/")) + ")";
    }
}

File: kotlin_solutions/553.optimal-division.kt
/*
 * @lc app=leetcode id=553 lang=kotlin
 *
 * [553] Optimal Division
 */

class Solution {fun optimalDivision(nums: IntArray) = if (nums.size == 1) "${nums[0]}" else if (nums.size == 2) "${nums[0]}/${nums[1]}" else "${nums[0]}/(${nums.slice(1 until nums.size).joinToString("/")})"}
